packages/access-api/src/routes/validate-email.js

```diff
@@ -134,6 +139,93 @@ async function session(req, env) {
     req.query.ucan,
     delegation.capabilities[0].nb.key
   )
+  if (req.method.toLowerCase() === 'post') {
+    const accessSessionResult = await validator.access(delegation, {
+      capability: Access.session,
+      principal: Verifier,
+      authority: env.signer,
+    })
+    if (accessSessionResult.error) {
+      throw new Error(
+        `unable to validate access session: ${accessSessionResult.error}`
+      )
+    }
+    const accountDID = accessSessionResult.audience.did()
+    // @todo: use new ucanto `Account` instead
+    const accountOneOffKey = await ed25519.generate()
+    const account = accountOneOffKey.withDID(accountDID)
+
+    const agentPubkey = accessSessionResult.capability.nb.key
+
+    const update = await ucanto.delegate({
+      issuer: env.signer,
+      audience: account,
+      capabilities: [
+        {
+          with: env.signer.did(),
+          can: './update',
+          nb: {
+            key: agentPubkey,
+          },
+        },
+      ],
+    })
+    const attestKeyAuthorizesAccount = await ucanto.delegate({
+      issuer: env.signer,
+      audience: { did: () => agentPubkey },
+      capabilities: [
+        {
+          with: env.signer.did(),
+          can: 'ucan/attest',
+          nb: {
+            proof: update.cid,
+          },
+        },
+      ],
+      proofs: [update],
+    })
+
+    // create delegations that should be claimable
+    const delegationAccountToKey = await ucanto.delegate({
+      issuer: account,
+      audience: {
+        did() {
+          return agentPubkey
+        },
+      },
+      capabilities: [
+        {
+          with: 'ucan:*',
+          can: '*',
+        },
+      ],
+    })
+    // generate a delegation to the key that we can save in
+    // models.delegations to be found by subsequent access/claim
+    // invocations invoked by the did:key
+    const delegateToKey = await ucanto.delegate({
+      issuer: env.signer,
+      audience: {
+        did() {
+          return agentPubkey
+        },
+      },
+      proofs: [delegationAccountToKey],
```

Comment:
I'm not sure what is this for ? This could either contain delegation from `did:web:web3.storage` to a service key, or if they are the same no proof is required.